@here/olp-sdk-core/lib/utils/TileKey.ts

```diff
@@ -0,0 +1,417 @@
+        const result = TileKey.fromRowColumnLevel(row, column, level);
+        result.m_mortonCode = quadKey64;
+        return result;
+    }
+
+    /**
+     * Creates a tile key from a heretile code string.
+     *
+     * The string can be created with [[toHereTile]].
+     *
+     * @param quadkey64 The string representation of the HERE tile key.
+     * @returns A new instance of `TileKey`.
+     */
+    static fromHereTile(quadkey64: string): TileKey {
+        const result = TileKey.fromMortonCode(parseInt(quadkey64, 10));
+        result.m_hereTile = quadkey64;
+        return result;
+    }
+
+    /**
+     * Returns the number of available columns at a given level.
+     *
+     * This is 2 to the power of the level.
+     *
+     * @param level The level for which to return the number of columns.
+     * @returns The available columns at the given level.
+     */
+    static columnsAtLevel(level: number): number {
+        return Math.pow(2, level);
+    }
+
+    /**
+     * Returns the number of available rows at a given level.
+     *
+     * This is 2 to the power of the level.
+     *
+     * @param level The level for which to return the number of rows.
+     * @returns The available rows at the given level.
+     */
+    static rowsAtLevel(level: number): number {
+        return Math.pow(2, level);
+    }
+
+    /**
+     * Computes the Morton code of the parent tile key of the given Morton code.
+     *
+     * Note: The parent key of the root key is the root key itself.
+     *
+     * @param mortonCode A Morton code, for example, obtained from [[mortonCode]].
+     * @returns The Morton code of the parent tile.
+     */
+    static parentMortonCode(mortonCode: number): number {
+        return Math.floor(mortonCode / SUBTILES_COUNT);
+    }
+
+    /**
+     * Constructs a new immutable instance of a `TileKey`.
+     *
+     * For the better readability, [[TileKey.fromRowColumnLevel]] should be preferred.
+     *
+     * Note - row and column must not be greater than the maximum rows/columns for the given level.
+     *
+     * @param row Represents the row in the quadtree.
+     * @param column Represents the column in the quadtree.
+     * @param level Represents the level in the quadtree.
+     */
+    constructor(
+        readonly row: number,
+        readonly column: number,
+        readonly level: number
+    ) {}
+
+    /**
+     * Converts the tile key into a string for using in REST API calls.
+     *
+     * @returns The string is a number to the
+     * base of 4, but without the leading 1, with the following properties:
+     *  1. the number of digits equals the level.
+     *  2. removing the last digit gives the parent tile's quadkey string, i.e. appending 0,1,2,3
+     *     to a quadkey string gives the tiles's children.
+     *
+     * You can convert back from a quadkey string with [[fromQuadKey]].
+     */
+    public toQuadKeyString(): string {
+        let result = "";
```

Comment:
In C++ SDK there is the special case of "-" when TileKey is (0,0,0). It might be good to have it also here.

Reply:
updated